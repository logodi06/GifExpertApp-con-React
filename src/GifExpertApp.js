//Utilizar comando rafc para crear toda la estructura necesaria para la página

import React, { useState } from 'react'
import { AddCategory } from './components/AddCategory';
import { GifGrid } from './components/GifGrid';
 
/*export*/ const GifExpertApp = ( { defaultCategories  = [] }) => {

    //const categories = ['One Punch', 'Samurai X', 'Dragon Ball'];
    //const [categories, setcategories] = useState(['One Punch', 'Samurai X', 'Dragon Ball']);

    //const [categories, setcategories] = useState(['One Punch']);
    const [categories, setcategories] = useState(defaultCategories);

    /*const handleAdd = () => {
        //Operador xprect se utiliza para agregar elementos a un arreglo, manteniendo el arreglo
        //que se tenia anteriormente
        setcategories([...categories, 'XMEN']);

        //Otra forma de agregar al arreglo
        //setcategories(cats => [...cats, 'XMEN']);
    }*/


    return (
        <>
            <h2>GifExpertApp</h2>
            <AddCategory setcategories= {setcategories} />
            <hr/>
        
            <ol>
                {
                    //Metodo map, es un método que tienen los arreglos y sirve para 
                    //recorer los valores del arreglo
                    //EL map recibe 2 elementos, categoria y el indice
                    // categories.map( category => {
                    //     return <li key={category}> {category}</li> 
                    // })
                    categories.map( category => (
                        <GifGrid 
                            key={category}
                            category={category} 
                        />
                      ))
                }
            </ol>
        </>
    )
}

  
  export default GifExpertApp