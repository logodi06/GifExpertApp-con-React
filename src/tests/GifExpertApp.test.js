import { shallow } from "enzyme";
import GifExpertApp from "../GifExpertApp";

describe('Probar que el componente <GifExpertApp /> funcione correctamente', () => {

    test('Debe de mostrar correctamente el <GifExpertApp/>', () => {
        const wrapper = shallow(<GifExpertApp />);
        expect(wrapper).toMatchSnapshot();
    });

    test('Debe de mostrara una lista de categorias', () => {
        const categories = ['One Punch', 'Burger'];
        const wrapper = shallow(<GifExpertApp defaultCategories={categories}/>);
        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('GifGrid').length).toBe(categories.length);
    })
    
    
})
