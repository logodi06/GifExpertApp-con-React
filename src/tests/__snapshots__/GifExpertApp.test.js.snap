// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Probar que el componente <GifExpertApp /> funcione correctamente Debe de mostrar correctamente el <GifExpertApp/> 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setcategories={[Function]}
  />
  <hr />
  <ol />
</Fragment>
`;

exports[`Probar que el componente <GifExpertApp /> funcione correctamente Debe de mostrara una lista de categorias 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setcategories={[Function]}
  />
  <hr />
  <ol>
    <GifGrid
      category="One Punch"
      key="One Punch"
    />
    <GifGrid
      category="Burger"
      key="Burger"
    />
  </ol>
</Fragment>
`;
